backend/map/views.py

```diff
@@ -1,12 +1,17 @@
-from django.shortcuts import render
-from django.http import JsonResponse
-from rest_framework.decorators import api_view
-
-from .use_cases import (
-    get_user_all_locations_and_company
-)
-
-# map data
-def get_map_data(request):
-    user_id = request.session.get("user_id")  or '0'
-    return JsonResponse(get_user_all_locations_and_company(user_id), safe=False)
+from django.shortcuts import render
+from django.http import JsonResponse
+from rest_framework.decorators import api_view
+
+from .use_cases import (

```

Comment:
I think you should make the view into a class, and then have an AbstractMapUseCase object as an attribute. Then, you can have the views.py file only depend on AbstractMapUseCase, and in the urls.py file, you can pass in a concrete MapUseCase object. That's the formatting of the dashboard, transaction, or login refactor and I think it shows better dependency inversion because the views shouldn't depend on a concrete implementation